### PR TITLE
update crossplane-runtime to v1.15.2

### PR DESCRIPTION
**Pull request by haarchri**
### Description of your changes

Fixes #

I have:

- [ ] Read and followed Crossplane's [contribution process].
- [ ] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

**Comment by haarchri** [comment]
i know ;) thats why its a draft - but with this branch we can use provider-argocd in functions

File: cmd/provider/main.go [hunk 1]
```diff
@@ -48,16 +56,39 @@ func main() {
...
-		SyncPeriod:       syncPeriod,
+		// controller-runtime uses both ConfigMaps and Leases for leader
+		// election by default. Leases expire after 15 seconds, with a
+		// 10 second renewal deadline. We've observed leader loss due to
+		// renewal deadlines being exceeded when under high load - i.e.
+		// hundreds of reconciles per second and ~200rps to the API
+		// server. Switching to Leases only and longer leases appears to
+		// alleviate this.
+		LeaderElection:             *leaderElection,
+		LeaderElectionID:           "crossplane-leader-election-provider-aws",
```
Comment (janwillies) [inline cmd/provider/main.go:73]: should be `argocd`

**Comment by maximilianbraun** [comment]
I think if that will be merged, the crossplane-runtime update should work quite well. All the replace's in the mod-file could be removed…
https://github.com/argoproj/gitops-engine/pull/556

**Comment by MisterMX** [comment]
I have made the code compile by replacing `github.com/argoproj/gitops-engine` with https://github.com/asaf-erlich/gitops-engine/commit/1c9f565e644fb51cc9afc96116fd788ff70390b8 which is the WIP in https://github.com/argoproj/gitops-engine/pull/565.

Since we are not using the gitops-engine itself and only use the API types I think it is safe to merge this.

**Review by MisterMX — APPROVED** [review]
LGTM. I fixed the remaining issues myself. Thanks a lot @haarchri!

**Pull request merged**